internal/host/monitor.go

```diff
@@ -35,16 +28,29 @@ func (m *Manager) initMonitoringQueryGenerator() {
 				models.HostStatusPreparingSuccessful,
 				models.HostStatusInstalling,
 				models.HostStatusInstallingInProgress,
-				models.HostStatusInstalled,
 				models.HostStatusInstallingPendingUserAction,
 				models.HostStatusResettingPendingUserAction,
-				models.HostStatusCancelled, // for limited time, until log collection finished or timed-out
-				models.HostStatusError,     // for limited time, until log collection finished or timed-out
 			}
 
-			dbWithCondition := common.LoadTableFromDB(db, common.HostsTable, "status in (?)", monitorStates)
-			dbWithCondition = common.LoadClusterTablesFromDB(dbWithCondition, common.HostsTable)
-			dbWithCondition = dbWithCondition.Where("exists (select 1 from hosts where clusters.id = hosts.cluster_id)")
+			// monitor following states for limited time, until log collection finished or timed-out
+			monitorStatesUntilLogCollection := []string{
+				models.HostStatusCancelled,
+				models.HostStatusError,
+			}
+			logCollectionEndStates := []string{
+				string(models.LogsStateCompleted),
+				string(models.LogsStateTimeout),
+				string(models.LogsStateEmpty),
+			}
+
+			dbWithCondition := common.LoadClusterTablesFromDB(db)
```

Comment:
How is this working? How is this selecting hosts? Am I missing something here? :thinking:

Reply:
this query selects clusters, not hosts (then the hosts are embeded in the cluster object). I fought with gorm to understand what all this does.
Now you tell me that, I wonder if we select distinct clusters. I need to check that.

Reply:
By the why, I see we batch queries here: https://github.com/openshift/assisted-service/blob/21d36c5b6b81d480fbfdb39666175abdbf75c964/internal/common/monitor_query_generator.go#L41

Wouldn't it make sense to use a cursor instead? (I don't know how it works with gorm though)

Reply:
Oh, you're right - I got confused by the deleted `"exists (select 1 from hosts where clusters.id = hosts.cluster_id)"`. As we are using subqueries and selecting from clusters it'll be bound to be distinct, right?

Reply:
yep, no problems